### DISCUSSION TITLE
Wrong value Home Power Consumption?

### DISCUSSION THREAD

**Discussion by phlexss** [klaasnicolaas/home-assistant-glow#413] [Q&A]
Some questions from a new glow user;

I have installed glow successfully (I think) and I get values in Home Assistant, but the "Home Power Consumption" seems way to high. It almost seems like the decimal sign is in the wrong place. I get values like '159.492 W'.

Home Daily Energy and Home Lifetime Energy look fine; 8,441 kWh and 8,325 kWh

How can the power consumption be so high but the energy values be ok?

I have a 1000 imp/kWh meter so I'm trying to reduce the amount of sensor data but I don't understand how to do it. The docs talk about 'components/pulse_meter.yaml' but I can't find this file. I have HA running on a pi4. Or can I just add the config to configuration.yaml?

Another question I have is about the energy dashboard which I just started using; should I configure daily sensor or lifetime sensors there?

Any help is appreciated.

**Comment by phlexss** [comment]
Ok, I found out how to compile the code and I uncommented the lines related to the filter. But after installing this update the values are even weirder. Power consumption value is now even higher.

Is it normal to get these high power consumption values and what is the correct way to reduce the amount of sensor data?

Thanks!

**Comment by klaasnicolaas** [comment]
Did you first adjust the sensitivity of the diode on the photodiode board? Based on the amount of kWh I cannot say much whether something is going wrong or right (be sure to add some logs).

> should I configure daily sensor or lifetime sensors there?

It does not matter which of the 2 you use, both are suitable for the energy dashboard

**Comment by mattiariz** [comment]
I think I have the same problem too.
my reader sends me 647.715w but it's around 1300.
while I look at the log I see the correct value while the page updates for a fraction of a second.
when you write: 
**red settings 
state OFF** 
the value is reported correctly. 
when instead you write 
**red settings 
state off 
red settings** 
the incorrect value is reported

16:36:22	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.06 s).
16:36:22	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:22	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:22	[D]	[light:036]	
**'Red' Setting:
16:36:22	[D]	[light:047]	
  State: OFF
16:36:22	[D]	[light:036]	
'Red' Setting:**
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Total Energy': Sending state 0.11900 kWh with 3 decimals of accuracy
16:36:24	[D]	[light:036]	
'Red' Setting:
16:36:24	[D]	[light:047]	
  State: ON
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Power Consumption': Sending state 2038.76721 W with 0 decimals of accuracy
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Daily Energy': Sending state 3.04817 kWh with 3 decimals of accuracy
16:36:24	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.05 s).
16:36:24	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Total Energy': Sending state 0.12000 kWh with 3 decimals of accuracy
16:36:24	[D]	[light:036]	
'Red' Setting:
16:36:24	[D]	[light:036]	
'Red' Setting:
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Power Consumption': Sending state 689919.50000 W with 0 decimals of accuracy
16:36:24	[D]	[sensor:094]	
'Home Assistant Glow - Daily Energy': Sending state 3.06504 kWh with 3 decimals of accuracy
16:36:24	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.07 s).
16:36:24	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:24	[D]	[light:036]	
'Red' Setting:
16:36:24	[D]	[light:047]	
  State: OFF
16:36:24	[D]	[light:036]	
'Red' Setting:
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Total Energy': Sending state 0.12100 kWh with 3 decimals of accuracy
16:36:26	[D]	[light:036]	
'Red' Setting:
16:36:26	[D]	[light:047]	
  State: ON
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Power Consumption': Sending state 2183.67090 W with 0 decimals of accuracy
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Daily Energy': Sending state 3.06599 kWh with 3 decimals of accuracy
16:36:26	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.07 s).
16:36:26	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Total Energy': Sending state 0.12200 kWh with 3 decimals of accuracy
16:36:26	[D]	[light:036]	
'Red' Setting:
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Power Consumption': Sending state 667408.25000 W with 0 decimals of accuracy
16:36:26	[D]	[sensor:094]	
'Home Assistant Glow - Daily Energy': Sending state 3.08397 kWh with 3 decimals of accuracy
16:36:26	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.06 s).
16:36:26	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:26	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:26	[D]	[light:036]	
'Red' Setting:
16:36:26	[D]	[light:047]	
  State: OFF
16:36:26	[D]	[light:036]	
'Red' Setting:
16:36:27	[D]	[sensor:094]	
'Home Assistant Glow - Total Energy': Sending state 0.12400 kWh with 3 decimals of accuracy
16:36:27	[D]	[light:036]	
'Red' Setting:
16:36:27	[D]	[light:047]	
  State: ON
16:36:27	[D]	[sensor:094]	
'Home Assistant Glow - Power Consumption': Sending state 4355.73535 W with 0 decimals of accuracy
16:36:27	[D]	[sensor:094]	
'Home Assistant Glow - Daily Energy': Sending state 3.08586 kWh with 3 decimals of accuracy
16:36:27	[W]	[component:214]	
Component pulse_meter.sensor took a long time for an operation (0.07 s).
16:36:27	[W]	[component:215]	
Components should block for at most 20-30ms.
16:36:27	[D]	[light:036]	
'Red' Setting:
16:36:27	[D]	[light:047]	
  State: OFF

**Reply by klaasnicolaas** [reply]
Considering the timestamp, it sees 1 pulse as 2, which means you see a high power consumption. What **pulse rate** do you use? And have you first tried to better adjust the sensitivity with the controller on the photodiode PCB? If so, there should be some options in the `pulse_meter` component to adjust some settings (which should actually be documented in the docs).

**Reply by phlexss** [reply]
I can't seem to find any other documentation apart from the 'getting started' part on the main website and don't find any mention anywhere about adjusting the sensitivity of the diode.

Is this just a matter of turning the dial on the controller board of the diode and see what happens?

I also struggle to find any options in pulse_meter.yaml that can affect this.

My meter has a pulse rate of 1000 imp/kWh.

What I can't understand is how the power-consumption sensor can have these very high values but the daily and lifetime sensors are more or less ok (they are also too high but not in the same order of magnitude). Surely all 3 sensors use the same pulse data!?

Is there a way to change the values in pulse_meter.yaml directly on the Home Assistant server (in my case a pi4b) without having to re-compile and flash?

**Comment by phlexss** [comment]
I finally fixed it by uncommenting:

internal_filter: 100ms

In the 'Power Consumption' configuration.

I had to re-compile and flash because when I "Adopt" in Home Assistant and edit the configuration there I get errors about duplicate key and "ID button_reset_total redefined! Check button->1->id." etc. when I "Install"......and so far I have not figured out how to fix that.
